lua/telescope/finders.lua

```diff
@@ -46,12 +49,17 @@ function JobFinder:new(opts)
   --        vim.loop.new_pipe (stdin / stdout). stdout => filter pipe
   --        rg huge_search | fzf --filter prompt_is > buffer. buffer could do stuff do w/ preview callback
 
+  self.__index = self
   local obj = setmetatable({
     entry_maker = opts.entry_maker or make_entry.from_string,
     fn_command = opts.fn_command,
     static = opts.static,
     state = {},
 
+    track_results = opts.track_results,
```

Comment:
So it's not exactly about tracking results it seems, it's more that progressively narrowing results over time.

I'm not sure if this is 100% true for you, but can something not match the first parts of your prompt and then match later? I know for some of my sorters it is possible.